src/modules/ekf2/ekf2_main.cpp

```diff
@@ -1212,18 +1147,27 @@ void Ekf2::run()
 			}
 
 		} else {
-			_range_finder_sub_index = getRangeSubIndex(_range_finder_subs);
+			for (unsigned i = 0; i < ORB_MULTI_MAX_INSTANCES; i++) {
+				if (_range_finder_subs[i].updated()) {
+					distance_sensor_s report{};
+					_range_finder_subs[i].copy(&report);
+
+					// only use the first instace which has the correct orientation
+					if (report.orientation == distance_sensor_s::ROTATION_DOWNWARD_FACING) {
+						PX4_INFO("Found range finder with instance %d", i);
+						_range_finder_sub_index = i;
```

Comment:
I think having this encapsulated in a function was nicer, just give it a better name so it says what it does.

Reply:
Makes sense, restored.